File: GotlandBackend/Models/BookingPax.cs
using System;
using Accidis.WebServices.Models;

namespace Accidis.Gotland.WebService.Models
{
	public sealed class BookingPax
	{
		public Guid Id { get; set; }
		public Guid BookingId { get; set; }
		public string FirstName { get; set; }
		public string LastName { get; set; }
		public Gender Gender { get; set; }
		public DateOfBirth Dob { get; set; }
		public int CabinClassMin { get; set; }
		public int CabinClassPreferred { get; set; }
		public int CabinClassMax { get; set; }
		public string SpecialRequest { get; set; }

		public static BookingPax FromSource(BookingSource.PaxSource source, Guid bookingId)
		{
			return new BookingPax
			{
				BookingId = bookingId,
				FirstName = source.FirstName,
				LastName = source.LastName,
				Gender = Gender.FromString(source.Gender),
				Dob = new DateOfBirth(source.Dob),
				CabinClassMin = source.CabinClassMin,
				CabinClassPreferred = source.CabinClassPreferred,
				CabinClassMax = source.CabinClassMax,
				SpecialRequest = source.SpecialRequest
			};
		}
	}
}

File: GotlandFrontend/lib/model/booking_pax.dart
import 'json_field.dart';

class BookingPax {
	final String firstName;
	final String lastName;
	final String gender;
	final String dob;
	final String food;
	final int cabinClassMin;
	final int cabinClassPreferred;
	final int cabinClassMax;

	BookingPax(this.firstName, this.lastName, this.gender, this.dob, this.food, this.cabinClassMin, this.cabinClassPreferred, this.cabinClassMax);

	factory BookingPax.fromMap(Map<String, dynamic> map) =>
		BookingPax(
			map[FIRSTNAME],
			map[LASTNAME],
			map[GENDER],
			map[DOB],
			map[FOOD],
			map[CABIN_CLASS_MIN],
			map[CABIN_CLASS_PREF],
			map[CABIN_CLASS_MAX],
		);

	Map<String, dynamic> toMap() =>
		<String, dynamic>{
			FIRSTNAME: firstName,
			LASTNAME: lastName,
			GENDER: gender,
			DOB: dob,
			FOOD: food,
			CABIN_CLASS_MIN: cabinClassMin,
			CABIN_CLASS_PREF: cabinClassPreferred,
			CABIN_CLASS_MAX: cabinClassMax
		};
}
